Model/Resolver/CustomerOrders.php

```diff
@@ -0,0 +1,164 @@
+ */
+class CustomerOrders extends CoreCustomerOrders
+{
+    /**
+     * @var SearchCriteriaBuilder
+     */
+    protected $searchCriteriaBuilder;
+
+    /**
+     * @var OrderRepositoryInterface
+     */
+    protected $orderRepository;
+
+    /**
+     * @var OrderFilter
+     */
+    protected $orderFilter;
+
+    /**
+     * @var OrderFormatter
+     */
+    protected $orderFormatter;
+
+    /**
+     * @var SortOrderBuilder
+     */
+    protected $sortOrderBuilder;
+
+    /**
+     * CustomerOrders constructor.
+     * @param OrderRepositoryInterface $orderRepository
+     * @param SearchCriteriaBuilder $searchCriteriaBuilder
+     * @param OrderFilter $orderFilter
+     * @param OrderFormatter $orderFormatter
+     * @param SortOrderBuilder $sortOrderBuilder
+     */
+    public function __construct(
+        OrderRepositoryInterface $orderRepository,
+        SearchCriteriaBuilder $searchCriteriaBuilder,
+        OrderFilter $orderFilter,
+        OrderFormatter $orderFormatter,
+        SortOrderBuilder $sortOrderBuilder
+    ) {
+        parent::__construct(
+            $orderRepository,
+            $searchCriteriaBuilder,
+            $orderFilter,
+            $orderFormatter
+        );
+
+        $this->orderRepository = $orderRepository;
+        $this->searchCriteriaBuilder = $searchCriteriaBuilder;
+        $this->orderFilter = $orderFilter;
+        $this->orderFormatter = $orderFormatter;
+        $this->sortOrderBuilder = $sortOrderBuilder;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function resolve(
+        Field $field,
+        $context,
+        ResolveInfo $info,
+        array $value = null,
+        array $args = null
+    ) {
+        if ($context->getExtensionAttributes()->getIsCustomer() === false) {
+            throw new GraphQlAuthorizationException(__('The current customer isn\'t authorized.'));
+        }
+
+        if ($args['currentPage'] < 1) {
+            throw new GraphQlInputException(__('currentPage value must be greater than 0.'));
+        }
+
+        if ($args['pageSize'] < 1) {
+            throw new GraphQlInputException(__('pageSize value must be greater than 0.'));
+        }
+
+        $userId = $context->getUserId();
+        /** @var StoreInterface $store */
+        $store = $context->getExtensionAttributes()->getStore();
+
+        try {
+            $searchResult = $this->getSearchResult($args, (int)$userId, (int)$store->getId());
+            $maxPages = (int)ceil($searchResult->getTotalCount() / $searchResult->getPageSize());
+        } catch (InputException $e) {
+            throw new GraphQlInputException(__($e->getMessage()));
+        }
+
+        $ordersArray = [];
+
+        foreach ($searchResult->getItems() as $orderModel) {
+            $ordersArray[] = $this->orderFormatter->format($orderModel);
+        }
+
+        // Need to reverse order array so that new orders will be first, then old
+        $reversedOrdersArray = array_reverse($ordersArray);
```

Comment:
this doesn't make much sense, since the search criteria already gets the orders by entity_id DESC; why are we reversing the results?
it is paginated results, too, so you're reversing the results in the bounds of the current page only, anyway

Reply:
Basically issue was smthing like this
![image (1)](https://user-images.githubusercontent.com/68007919/145056453-3db52870-d0b1-411d-9861-25cf634f63c9.png)
:
![image](https://user-images.githubusercontent.com/68007919/145056464-4c23ff59-38f3-4a84-8027-d00bb4cc969a.png)

Default magento endpoint was taking orders in not correct way. Basically it was getting from start of table so on first page we could see old one orders and on second new one. And I reversed array because if we set sort order, it is actually making correct fetch in this way, but shows first old then new.

What you think, maybe there is another way to make it in right approach?

Reply:
If you added desc sorting, I assume you wouldn't need array reverse?
can you share what the SQL for the select is on the orderRepository's getList call, at the moment?

array reverse is certainly not correct since it only works in the bounds of the current page

I assume the desired result is to see
# 1373
# 1344
# 1249
...
on the first page?
the sorting in search criteria should be doing it without array_reverse, so there's some voodoo going on

Reply:
Yes, you completely right. If we set order DESC we already getting them in right way. On FE old code was reversing it while formatting. 

Thanks for a good catch !